wizard.py

```diff
@@ -0,0 +1,1038 @@
+          enabled = step.get(inspect_type)
+          val = step.get(inspect_type + "_value")
+          operator = step.get(inspect_type + "_operator")
+
+          # Construct return value or byproducts inspection command
+          # c.f. https://github.com/in-toto/in-toto-inspections
+          if enabled:
+            # Inspection commands require a link file (to inspect)
+            # We use the auto name of the according QA step to
+            # generate the filename passed to the inspection command
+            # FIXME: What about the functionary keyid in the link file name?!
+            link = in_toto.models.link.FILENAME_FORMAT_SHORT.format(
+                step_name=step_name)
+            value = step.get(inspect_type + "_value")
+
+            if inspect_type == "retval":
+              run = ("inspect-return-value --link={link} --{operator} {value}"
+                  .format(link=link, operator=operator, value=value))
+
+            elif inspect_type in ["stdout", "stderr"]:
+              if operator == "empty":
+                operator = "is"
+                value = ""
+
+              run = ("inspect-byproducts"
+                  " --link={link} --{inspect_type} --{operator} \"{value}\""
+                  .format(link=link, inspect_type=inspect_type,
+                  operator=operator, value=value))
+
+            # FIXME: Come up with better auto names
+            inspect_name = "inspection-" + str(len(ssc_inspections) + 1)
+            ssc_inspections.append({
+              "name": inspect_name,
+              "cmd": run,
+              "based_on": step_name
+            })
+
+      ssc_steps.append(ssc_step)
+
+  ssc_data = {
+    "steps": ssc_steps,
+    "inspections": ssc_inspections
+  }
+  return ssc_data
+
+
+def form_data_to_ssc(step_names, step_commands, step_modifies,
+    inspection_names, inspection_commands, inspection_step_names):
+  """
+  <Purpose>
+    Takes step and inspection properties (form posted lists) and aggregates
+    them by index to step and inspection dictionaries to construct a dictionary
+    of software supply chain (ssc) data.
+
+  <Returns>
+    Software Supply Chain data, i.e. a dictionary of step and inspection data
+    Note: these are not not actual in-toto Step and Inspection objects
+
+    e.g.:
+    {
+      steps: [
+        {
+          "name": <unique step name>,
+          "cmd": <expected command>
+          "modifies": <boolean>  # whether the step modifies artifacts or not
+        }, ...
+      ],
+      inspections: [
+        {
+          "name": <unique inspection name>,
+          "cmd": <command to run inspecting>,
+          "based_on": <step name whose products are used for that inspection>
+        }
+      ]
+    }
+
+  """
+  steps = []
+  for i in range(len(step_names)):
+    steps.append({
+        "name": step_names[i],
+        "cmd": step_commands[i],
+        "modifies": step_modifies[i] == "true"
+      })
```

Comment:
we probably want to check step_{names,commands,modifies} is of the same length.

Reply:
+1 for defensive programming. Currently I only check the lengths [before calling the function](https://github.com/in-toto/layout-web-tool/blob/1bbbc531cc1cf30981677beaa575461b7f6761fe/wizard.py#L548-L553). Will change.

Reply:
Fixed with 131027476205f0a3c3dc620b4722a9e276355c65